--- sqlparse/sql.py.orig	2012-03-29 12:09:46.660311410 +0200
+++ sqlparse/sql.py	2012-03-29 12:11:04.924699486 +0200
@@ -23,17 +23,13 @@
         self.parent = None
 
     def __str__(self):
-        return str(self).encode('utf-8')
+        return self.value or ''
 
     def __repr__(self):
         short = self._get_repr_value()
         return '<%s \'%s\' at 0x%07x>' % (self._get_repr_name(),
                                           short, id(self))
 
-    def __unicode__(self):
-        """Returns a unicode representation of this object."""
-        return self.value or ''
-
     def to_unicode(self):
         """Returns a unicode representation of this object.
         
@@ -139,7 +135,7 @@
         self.tokens = tokens
         Token.__init__(self, None, None)
 
-    def __unicode__(self):
+    def __str__(self):
         return ''.join(str(x) for x in self.flatten())
 
     def _get_repr_name(self):
